### PR TITLE
Khaines/improved redis options

### PR DESCRIPTION
**Pull request by khaines**
**What this PR does**:  Enhancement to the redis cache package to support specifying a password and enabling TLS communication, via config. Enables the use of hosted Redis services such as Microsoft Azure Cache for Redis.

**Checklist**
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 * [FEATURE] The distributor can now drop labels from samples (similar to the removal of the replica label for HA ingestion) per user via the `distributor.drop-label` flag. #1726
 * [FEATURE] Added `global` ingestion rate limiter strategy. Deprecated `-distributor.limiter-reload-period` flag. #1766
 * [FEATURE] Added support for Microsoft Azure blob storage to be used for storing chunk data. #1913
+* [ENHANCEMENT] Added `password` and `enableTLS` options to redis cache configuration. Enables usage of Microsoft Azure Cache for Redis service.
```
Comment (pracucci) [inline CHANGELOG.md]: `enableTLS` > `enable_tls`

File: pkg/chunk/cache/redis_cache.go [hunk 2]
```diff
@@ -34,6 +36,8 @@ func (cfg *RedisConfig) RegisterFlagsWithPrefix(prefix, description string, f *f
 	f.DurationVar(&cfg.Expiration, prefix+"redis.expiration", 0, description+"How long keys stay in the redis.")
 	f.IntVar(&cfg.MaxIdleConns, prefix+"redis.max-idle-conns", 80, description+"Maximum number of idle connections in pool.")
 	f.IntVar(&cfg.MaxActiveConns, prefix+"redis.max-active-conns", 0, description+"Maximum number of active connections in pool.")
+	f.StringVar(&cfg.Password, prefix+"redis.password", "", "password to use when connecting to redis.")
```
Comment (pracucci) [inline pkg/chunk/cache/redis_cache.go]: Please add `description+` to the help message and starts the help message with a capital letter, due to how the final message is composed.

_Same feedback applies to the next CLI flag._

File: CHANGELOG.md (hunk 1 above)
Comment (khaines) [reply CHANGELOG.md]: good catch. corrected

File: pkg/chunk/cache/redis_cache.go (hunk 2 above)
Comment (khaines) [reply pkg/chunk/cache/redis_cache.go]: Thanks @pracucci . Fixed this.

**Pull request merged**